packages/auth/src/grant/routes.ts

```diff
@@ -96,17 +96,26 @@ async function createGrantInitiation(
       })
       .every((el) => el === true)
   ) {
-    const grant = await grantService.create(body)
-    const accessToken = await deps.accessTokenService.create(grant.id)
-    const access = await deps.accessService.getByGrant(grant.id)
-    ctx.status = 200
-    ctx.body = createGrantBody({
-      domain: deps.config.authServerDomain,
-      grant,
-      access,
-      accessToken
-    })
-    return
+    const trx = await Grant.startTransaction()
+    try {
+      const grant = await grantService.create(body, trx)
+      const accessToken = await deps.accessTokenService.create(grant.id, {
+        trx
+      })
+      await trx.commit()
+      const access = await deps.accessService.getByGrant(grant.id)
+      ctx.status = 200
+      ctx.body = createGrantBody({
+        domain: deps.config.authServerDomain,
+        grant,
+        access,
+        accessToken
+      })
+      return
+    } catch (err) {
+      await trx.rollback()
```

Comment:
Does this actually roll back the transaction if we commit in line 105?

Reply:
:thinking: yeah, I'm not sure what'd happen if it threw after the commit and tried to rollback.
Options seem to be:

1. end the try block directly after the commit. (Declare `let access: Access[]` before the try block, or return the access on the grant from `grantService.create` (and declare let grant: Grant))
2. Do this (instead of `commit`/`rollback`): https://github.com/wilsonianb/rafiki/pull/134/commits/726f759baea8eba17f74d5f1de18960a48c4b1ae But then, either `accessService.getByGrant` needs to accept a `trx` or you return `access` with the created `grant`.